notebooks/humans_or_horses.livemd

```diff
@@ -105,38 +175,42 @@ Axon.dropout and Axon.spatial_dropout create dropout layers which prevent the ne
 
 ```elixir
-defp build_model(input_shape, transpose_shape) do
-  Axon.input("input", shape: input_shape)
-  |> Axon.transpose(transpose_shape)
-  |> Axon.conv(16, kernel_size: {3, 3}, activation: :relu)
-  |> Axon.max_pool(kernel_size: {2, 2})
-  |> Axon.conv(32, kernel_size: {3, 3}, activation: :relu)
-  |> Axon.spatial_dropout(rate: 0.5)
-  |> Axon.max_pool(kernel_size: {2, 2})
-  |> Axon.conv(64, kernel_size: {3, 3}, activation: :relu)
-  |> Axon.spatial_dropout(rate: 0.5)
-  |> Axon.max_pool(kernel_size: {2, 2})
-  |> Axon.conv(64, kernel_size: {3, 3}, activation: :relu)
-  |> Axon.max_pool(kernel_size: {2, 2})
-  |> Axon.conv(64, kernel_size: {3, 3}, activation: :relu)
-  |> Axon.max_pool(kernel_size: {2, 2})
-  |> Axon.flatten()
-  |> Axon.dropout(rate: 0.5)
-  |> Axon.dense(512, activation: :relu)
-  |> Axon.dense(2, activation: :softmax)
+defmodule HorsesHumans.Model do
+  def build_model(input_shape, transpose_shape) do
+    Axon.input(input_shape, "input")
+    |> Axon.transpose(transpose_shape)
+    |> Axon.conv(16, kernel_size: {3, 3}, activation: :relu)
+    |> Axon.max_pool(kernel_size: {2, 2})
+    |> Axon.conv(32, kernel_size: {3, 3}, activation: :relu)
+    |> Axon.spatial_dropout(rate: 0.5)
+    |> Axon.max_pool(kernel_size: {2, 2})
+    |> Axon.conv(64, kernel_size: {3, 3}, activation: :relu)
+    |> Axon.spatial_dropout(rate: 0.5)
+    |> Axon.max_pool(kernel_size: {2, 2})
+    |> Axon.conv(64, kernel_size: {3, 3}, activation: :relu)
+    |> Axon.max_pool(kernel_size: {2, 2})
+    |> Axon.conv(64, kernel_size: {3, 3}, activation: :relu)
+    |> Axon.max_pool(kernel_size: {2, 2})
+    |> Axon.flatten()
+    |> Axon.dropout(rate: 0.5)
+    |> Axon.dense(512, activation: :relu)
+    |> Axon.dense(2, activation: :softmax)
+  end
 end
 ```
 
 ## Training
 
-Creating a training loop starts with setting a loss function to binary cross-entropy and optimizer of your choice. We also set the log option to 1 to keep track of loss and metrics during training. Next, we choose accuracy as a metric. Finally, we pass the number of epochs and set the number of iterations per epoch to 100.
+Creating a training loop starts with setting a loss function to binary cross-entropy and optimizer of your choice. We also set the log option to 1 to keep track of loss and metrics during training. Next, we choose accuracy as a metric. Finally, we pass the number of epochs and set the number of iterations per epoch to 50.
 
 ```elixir
-defp train_model(model, data, optimizer, epochs) do
-  model
-  |> Axon.Loop.trainer(:binary_cross_entropy, optimizer, log: 1)
-  |> Axon.Loop.metric(:accuracy)
-  |> Axon.Loop.run(data, %{}, epochs: epochs, iterations: 100)
+defmodule HorsesHumans.Training do
+  def train_model(model, data, optimizer, epochs) do
+    model
+    |> Axon.Loop.trainer(:binary_cross_entropy, optimizer, log: 1)
```

Comment:
Axon's BCE is specifically designed for the case of an output with a single probability between 0 and 1---I would replace this with `:categorical_cross_entropy`.